kubernetes/client.go

```diff
@@ -130,7 +130,13 @@ func getConfigForLocalCluster() (*rest.Config, error) {
 	remoteSecretPath := kialiconfig.Get().Deployment.RemoteSecretPath
 	if remoteSecret, readErr := GetRemoteSecret(remoteSecretPath); readErr == nil {
 		log.Debugf("Using remote secret for local cluster config found at: [%s]. Kiali must be running outside the kube cluster.", remoteSecretPath)
-		return clientcmd.NewDefaultClientConfig(*remoteSecret, nil).ClientConfig()
+		cc, err := clientcmd.NewDefaultClientConfig(*remoteSecret, nil).ClientConfig()
+		if err != nil {
+			return cc, err
+		}
+		cc.BearerToken = "" // force use of BearerTokenFile so client can refresh the token
+		cc.BearerTokenFile = remoteSecretPath
```

Comment:
I don't think you need to do this for the remote secrets. They will use long lived tokens: https://github.com/kiali/kiali/blob/master/hack/istio/multicluster/kiali-prepare-remote-cluster.sh#L127-L141 and I don't think those can be managed or refreshed by client-go since it is for a service account in another cluster.